src/NServiceBus.AcceptanceTests/DelayedDelivery/TimeoutManager/When_timeout_dispatch_fails.cs

```diff
@@ -0,0 +1,124 @@
+                        options.RouteToLocalEndpointInstance();
+                        return bus.Send(new MyMessage(), options);
+                    }))
+                    .WithEndpoint<ErrorSpy>()
+                    .Done(c => c.FailedTimeoutMovedToError)
+                    .Repeat(r => r.For<AllTransportsWithoutNativeDeferral>())
+                    .Should(c => Assert.AreEqual(5, c.NumTimesStorageCalled))
+                    .Run();
+        }
+
+        public class Context : ScenarioContext
+        {
+            public bool FailedTimeoutMovedToError { get; set; }
+            public int NumTimesStorageCalled { get; set; }
+        }
+
+        public class Endpoint : EndpointConfigurationBuilder
+        {
+            public Endpoint()
+            {
+                EndpointSetup<DefaultServer>(config =>
+                {
+                    config.EnableFeature<TimeoutManager>();
+                    config.UsePersistence<FakeTimeoutPersistence>();
+                    config.SendFailedMessagesTo(ErrorQueueForTimeoutErrors);
+                    config.RegisterComponents(c => c.ConfigureComponent<FakeTimeoutStorage>(DependencyLifecycle.SingleInstance));
+                });
+            }
+
+            class FakeTimeoutPersistence : PersistenceDefinition
+            {
+                public FakeTimeoutPersistence()
+                {
+                    Supports<StorageType.Timeouts>(s => { });
+                }
+            }
+
+            class FakeTimeoutStorage : IQueryTimeouts, IPersistTimeouts
+            {
+                public Context TestContext { get; set; }
+                
+                public Task<TimeoutsChunk> GetNextChunk(DateTime startSlice)
+                {
+                    var timeout = new TimeoutsChunk.Timeout(Guid.NewGuid().ToString(), DateTime.UtcNow);
+                    var timeouts = new List<TimeoutsChunk.Timeout>() { timeout };
+                    return Task.FromResult(new TimeoutsChunk(timeouts, DateTime.UtcNow + TimeSpan.FromSeconds(10)));
+                }
+
+                public Task Add(TimeoutData timeout, ContextBag context)
+                {
+                    return Task.FromResult(0);
+                }
+
+                public Task<bool> TryRemove(string timeoutId, ContextBag context)
+                {
+                    throw new NotImplementedException();
+                }
+
+                public Task<TimeoutData> Peek(string timeoutId, ContextBag context)
+                {
+                    TestContext.NumTimesStorageCalled++;
+                    throw new Exception("Ooops. Timeout storage went bust");
+                }
+
+                public Task RemoveTimeoutBy(Guid sagaId, ContextBag context)
+                {
+                    throw new NotImplementedException();
+                }
+            }
+        }
+
+        public class ErrorSpy : EndpointConfigurationBuilder
```

Comment:
@MarcinHoppe there is an extension to extract the exception, have a look at `When_auto_correlated_property_is_changed` for example.

I don't like the `ErrorSpy` approach since it spins up a second endpoint. 

> It also allows me to better verify at this endpoint that it is actually the message I expect that was delivered at the error queue

The spy in this test doesn't do any verification besides the message type. Also it does not verify WHY the message has failed.